lib/vanagon/project.rb

```diff
@@ -162,12 +162,20 @@ def merged_environment
     # Collects all sources and patches into the provided workdir
     #
     # @param workdir [String] directory to stage sources into
-    def fetch_sources(workdir)
+    # @param retry_count [Number] number of times to retry each fetch
+    # @param timeout [Number] How long to wait for each fetch before aborting
+    def fetch_sources(workdir, retry_count = 1, timeout = 7200)
       @components.each do |component|
-        component.get_source(workdir)
+        Vanagon::Utilities.retry_with_timeout(retry_count, timeout) do
+          component.get_source(workdir)
+        end
         # Fetch secondary sources
-        component.get_sources(workdir)
-        component.get_patches(workdir)
+        Vanagon::Utilities.retry_with_timeout(retry_count, timeout) do
+          component.get_sources(workdir)
```

Comment:
was this meant to be `get_patches`? you've got `get_sources` twice.

Reply:
oh wait, that's below..

Reply:
Not the best-named methods in the world, alas

Reply:
`get_source` vs `get_sources` ugh

Reply:
ignore me

Reply:
![ignore me -grand galactic inquisitor](https://user-images.githubusercontent.com/344926/34793596-cf020300-f600-11e7-852c-7960ab333fd2.jpg)

For future use